### ISSUE TITLE
RedisChannelLayer.send channel capacity checks against real internal redis queue, instead of expected asgi channel

### ISSUE THREAD

**Issue by thomas-riccardi** [django/channels_redis#97]
ASGI Channels for the same `RedisChannelLayer` instance (https://github.com/django/channels_redis/blob/54d935beb6842333ba4f3e6c29cbe86c4307cf16/channels_redis/core.py#L239) are factorized into one redis queue (also sometimes called a channel, which is confusing).
The ASGI channels capacity is check against the internal redis queue length, not the asgi channel.

Related to #95.

It is not only an issue if we have different rules per channel, but also and primarily an issue if we have many asgi channels for one `RedisChannelLayer` instance: some asgi channels may get `ChannelFull` because of other asgi channels being non-empty.

I'm not sure what to do to fix this, but it could at least be documented in the README (https://github.com/django/channels_redis#channel_capacity).

**Comment by andrewgodwin** [comment]
Documenting this in the README is the correct approach I feel. I don't think there's any other sensible way around this without throwing away the significant efficiency gains of reading from a single list.

**Comment by thomas-riccardi** [comment]
Do you propose to stop channel-specific capacity altogether ? (we should then remove the examples from the README)
Indeed I don't see any other reasonable way.

**Comment by andrewgodwin** [comment]
No, the backpressure is valuable to stop Redis filling up, and I think that part still works fine if it's shared across channels.
